From 5910be447734cfc19dccd7ce7cf5e9d053f28441 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:59:20 -0400
Subject: [fs] btrfs: don't run delayed nodes again after all nodes flush

Message-id: <1407179109-13714-74-git-send-email-sandeen@redhat.com>
Patchwork-id: 87047
O-Subject: [RHEL7.1 PATCH 073/422] Btrfs: don't run delayed nodes again after all nodes flush
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Miao Xie <miaox@cn.fujitsu.com>

commit 4dd466d36a50d9c96cd990e2e0d472fe720a10aa
Author: Miao Xie <miaox@cn.fujitsu.com>
Date:   Thu Dec 26 13:07:02 2013 +0800

    Btrfs: don't run delayed nodes again after all nodes flush

    If the number of the delayed items is greater than the upper limit, we will
    try to flush all the delayed items. After that, it is unnecessary to run
    them again because they are being dealt with by the wokers or the number of
    them is less than the lower limit.

    Signed-off-by: Miao Xie <miaox@cn.fujitsu.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/delayed-inode.c b/fs/btrfs/delayed-inode.c
index d68dc41..fe1c275 100644
--- a/fs/btrfs/delayed-inode.c
+++ b/fs/btrfs/delayed-inode.c
@@ -1395,6 +1395,7 @@ void btrfs_balance_delayed_items(struct btrfs_root *root)
     break;
   }
   finish_wait(&delayed_root->wait, &__wait);
+  return;
  }
 
  btrfs_wq_run_delayed_node(delayed_root, root, BTRFS_DELAYED_BATCH);
-- 
1.7.1